core/src/main/scala/caliban/schema/SchemaUtils.scala

```diff
@@ -4,8 +4,15 @@ import caliban.ResponseValue.ObjectValue
 import caliban.Value.{ EnumValue, NullValue, StringValue }
 import caliban.introspection.adt.{ __DeprecatedArgs, __Field, __Type }
 import caliban.schema.Step.MetadataFunctionStep
+import caliban.parsing.adt.Directive
 
 private[schema] object SchemaUtils {
+
+  /**
+   * Directive used to mark a field as semantically non-nullable.
+   */
+  val SemanticNonNull = Directive("semanticNonNull")
```

Comment:
Perhaps this is better placed in `caliban.parsing.adt.Directive`

Reply:
Since those are all name strings instead of actual directive instances, I'm worried that adding this causes some confusion :thinking: